src/test/java/com/cloudbees/hudson/plugins/folder/FolderTest.java

```diff
@@ -278,6 +286,32 @@ private void copyFromGUI(Folder f, JenkinsRule.WebClient wc, String fromName, St
             }
         });
     }
+    
+    @Issue("JENKINS-32359")
+    @Test public void shouldProperlyPersistFolderPropertiesOnMultipleReloads() throws Exception {
+        Folder folder = r.jenkins.createProject(Folder.class, "myFolder");
+        r.jenkins.setAuthorizationStrategy(new ProjectMatrixAuthorizationStrategy());
+        
+        // We add a stub property to generate the persisted list
+        // After that we save and reload the config in order to drop PersistedListOwner according to the JENKINS-32359 scenario
+        folder.addProperty(new FolderCredentialsProvider.FolderCredentialsProperty(new DomainCredentials[0]));
+        folder.doReload();
+        
+        // Add another property
+        Map<Permission,Set<String>> grantedPermissions = new HashMap<Permission, Set<String>>();
+        Set<String> sids = new HashSet<String>();
+        sids.add("admin");
+        grantedPermissions.put(Jenkins.ADMINISTER, sids);
+        folder.addProperty(new com.cloudbees.hudson.plugins.folder.properties.AuthorizationMatrixProperty(grantedPermissions));
+        
+        // Reload folder from disk and check the state
+        folder.doReload();
+        Folder reloadedFolder = r.jenkins.getItemByFullName("myFolder", Folder.class);
+        assertThat("Folder has not been found after the reloading", reloadedFolder, notNullValue());
+        assertThat("Property has not been reloaded, hence it has not been saved properly",
+            reloadedFolder.getProperties().get(com.cloudbees.hudson.plugins.folder.properties.AuthorizationMatrixProperty.class),
+            notNullValue());
```

Comment:
Would be nice to also verify that the `AbstractFolderProperty.owner` is set on all properties after reload.

Since that is `protected`, and not trivially accessed from implementations in this plugin, may be easier to use `@TestExtension` properties for this purpose.